src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -197,6 +198,7 @@ stop_metadata_sync_to_node(PG_FUNCTION_ARGS)
 
 	text *nodeName = PG_GETARG_TEXT_P(0);
 	int32 nodePort = PG_GETARG_INT32(1);
+	bool clearMetadata = PG_GETARG_BOOL(2);
 	char *nodeNameString = text_to_cstring(nodeName);
 
```

Comment:
In `start_metadata_sync_to_node()`, we prevent the command to run in a transaction block:
```SQL
BEGIN;
SELECT start_metadata_sync_to_node(nodename,nodeport) FROM pg_dist_node;
ERROR:  start_metadata_sync_to_node cannot run inside a transaction block
Time: 2.061 ms
```

I think running inside a transaction block with `SendOptionalCommandListToWorkerInTransaction` or `SendCommandListToWorkerInSingleTransaction ` APIs are not possible. Because, those functions commit on the workers immediately, preventing any `ROLLBACK` opportunity. 


so, we should do similar to `start_metadata_sync_to_node` to prevent it:
```C
	PreventInTransactionBlock(true, "start_metadata_sync_to_node");
```